### PR TITLE
Adding terminology for magic literals

### PR DESCRIPTION
**Pull request by gbrandtio**
Terminology for magic literals might be beneficial, in a sense that will make a strong impression to people and will be remembered when writing/reviewing code. 

Also allows easier communication between peers, since the terminology used is an industry standard (everybody will know what the discussion is about, when it comes to _magic literals_, _magic numbers_, _magic strings_).

### CODE REVIEW

**Comment by gbrandtio** [comment]
@dimadeveatii could you kindly review this 😄 ?

File: README.md [hunk 1]
```diff
@@ -131,7 +131,7 @@ setTimeout(restart, 86400000);
 **Good:**
 
 ```ts
-// Declare them as capitalized named constants.
+// Declare them as capitalized symbolic constants.
```
Comment (dimadeveatii) [inline README.md]: How is _symbolic constants_ better than _named constants_? The use of term _symbolic_ might confuse people to use `Symbol`, which is not the intention.

File: README.md [hunk 2]
```diff
@@ -117,9 +117,9 @@ function getUser(): User;
 
 **[⬆ back to top](#table-of-contents)**
 
-### Use searchable names
+### Use searchable names / Avoid Magic Literals
 
-We will read more code than we will ever write. It's important that the code we do write must be readable and searchable. By *not* naming variables that end up being meaningful for understanding our program, we hurt our readers. Make your names searchable. Tools like [ESLint](https://typescript-eslint.io/) can help identify unnamed constants.
+We will read more code than we will ever write. It's important that the code we do write must be readable and searchable. By *not* naming variables that end up being meaningful for understanding our program, we hurt our readers. Make your names searchable. Tools like [ESLint](https://typescript-eslint.io/) can help identify magic literals.
```
Comment (dimadeveatii) [inline README.md]: The term _searchable names_ is just what it says and it's easy to understand for a beginner who's not yet comfortable with all of the programming dictionary. While you say _magic literals_, others call them [_magic strings_](https://deviq.com/antipatterns/magic-strings).  

I consider the term _searchable names_ less confusing and easier to understand for someone who is a beginner.  
I do appreciate your contribution, and I'd suggest that we add it to the end of the paragraph instead:

> ...can help identify unnamed constants (also known as _magic literals_ and _magic strings_).

**Review by dimadeveatii — CHANGES_REQUESTED** [review]
Nice one 👍 . Please see my comments above.

File: README.md (hunk 2 above)
Comment (dimadeveatii) [reply README.md]: And in the end, the name _searchable names_ is referenced in the original _Clean Code_ book (see Chapter 2, page 22 - _Use Searchable Names_).

File: README.md (hunk 1 above)
Comment (gbrandtio) [reply README.md]: The term _symbolic constants_ is widely used throughout SE. However, seeing your next comment I do understand that the intention is to have a more straight-forward way for beginners to better get up to speed.

Is there any way we can incorporate both ? Something like the below for instance (might also help someone do further research on the term / get more familiar with):

```
// Declare them as capitalized symbolic constants (also known as symbolic constants).
```

File: README.md (hunk 2 above)
Comment (gbrandtio) [reply README.md]: Good point - the terminology for _magic literals_ referes to both _magic numbers_ and _magic strings_. Indeed though someone might be overwhelmed by this terminology. 

Addressed this as per the above comments, explicitly mentioning _magic numbers_ and _magic strings_ and bringing back the _unnamed constants_ on https://github.com/labs42io/clean-code-typescript/pull/59/commits/d1e2cfb632190e6990864ce64e81ce395b86b7d5

File: README.md (hunk 1 above)
Comment (dimadeveatii) [reply README.md]: The term _symbolic constant_ is more  oftenly used in C/C++, you would rarely see it in a JS/TS reference. It is also mentioned as _literal constant_ in other places. I prefer to keep it simple and not confuse the reader with too much terminology.

Comment (gbrandtio) [reply README.md]: Addressed on 36d14bf048c083c2a01b3c448f9a3c9bab80757e

**Pull request merged**